pkg/resolve/resolve.go

```diff
@@ -45,8 +46,14 @@ func ImageReferences(input []byte, builder build.Interface, publisher publish.In
 		}
 		// This simply returns the replaced object, which we discard during the gathering phase.
 		if _, err := replaceRecursive(obj, func(ref string) (string, error) {
-			if builder.IsSupportedReference(ref) {
-				refs[ref] = struct{}{}
+			if strict && !strings.HasPrefix(ref, "ko://") {
+				return ref, nil
+			}
+			tref := strings.TrimPrefix(ref, "ko://")
+			if builder.IsSupportedReference(tref) {
+				refs[tref] = struct{}{}
+			} else if strict && strings.HasPrefix(ref, "ko://") {
+				return "", fmt.Errorf("Strict reference %q is not supported", ref)
```

Comment:
"is not supported" makes sense from our go interfaces, but isn't particularly helpful from a CLI UX perspective.

You could cheat here and just call `builder.Build(tref)` and return whatever error it returns? I'm actually not sure what would happen but it's worth trying.

Reply:
Good feedback, changed to:

```
2019/08/15 15:01:51 error processing import paths in "test.yaml": Found strict reference "ko://github.com/blargh/whatever" but github.com/blargh/whatever is not a valid import path
```